### PR TITLE
[stable/prometheus] Add Namespace to applicable Resources

### PR DESCRIPTION
**Pull request by FrediWeber**
#### What this PR does / why we need it:
Add the namespace field to applicable resources. Otherwise `helm template` doesn't automatically add this field and the resulting resources are incomplete.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x ] Chart Version bumped
- [x ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by zanhsieh** [comment]
@FrediWeber 
Thanks for contribution. Just curious, how would this PR handle if the user would like to use install prometheus chart directly to specific namespace via cli?

**Comment by FrediWeber** [comment]
@zanhsieh 
Thank you for this question. Yes, this is a problem indeed.
But if you use the Kube State Metrics or Grafana component/chart for example, this problem already exists.

I think it's a new behavior and I see three ways to minimize the impact:
1. Inform the users somehow, that `helm template` is now behaving correctly, if you pass a namespace parameter to the `helm template` command but has the default namespace in it, if not.
2. Introduce a new parameter to this chart (for example `force-namespace`). The "old" behavior would still be the same but it would be possible to actually have namespaces with `helm template`.
3. Decide that we don't want this at all.

**Comment by zanhsieh** [comment]
> @zanhsieh
> Thank you for this question. Yes, this is a problem indeed.
> But if you use the Kube State Metrics or Grafana component/chart for example, this problem already exists.
> 
> I think it's a new behavior and I see three ways to minimize the impact:
> 
> 1. Inform the users somehow, that `helm template` is now behaving correctly, if you pass a namespace parameter to the `helm template` command but has the default namespace in it, if not.
> 2. Introduce a new parameter to this chart (for example `force-namespace`). The "old" behavior would still be the same but it would be possible to actually have namespaces with `helm template`.
> 3. Decide that we don't want this at all.
> 
> Here are some PRs from other charts with the exact same issue:
> 
> * #19704
> * #17689
> * #18990
> * #18993
> * #13924

I would pick option 2 since prometheus got integrated with other charts long time ago. Please make the change and bump PR to the latest version. Thank you.

/hold cancel

**Comment by FrediWeber** [comment]
@zanhsieh 
Okay, made the changes as requested. Thank you.

**Comment by FrediWeber** [comment]
/hold

I found some inconsistencies in my PR. Please wait with the merge.

**Comment by FrediWeber** [comment]
I reviewed this problem further and came to the conclusion, that it was not possible, to template this chart without `--namespace` and deploy it to a different namespace with `kubectl apply -n othernamespace ...`. This is because there are Clusterrolebindings in almost all aspects of this chart. Kubernetes requires to specify a namespace in the `subjects` section of Clusterrolebindings.
So even when other charts were to include this chart as a dependency and wanted to deploy this chart in another namespace than the parent chart - this would not work correctly because of the Clusterrolebindings.

https://github.com/helm/charts/blob/25ab41b4298b9253dfa3448efbf8577035f7fdc5/stable/prometheus/templates/server-clusterrolebinding.yaml#L11
https://github.com/helm/charts/blob/25ab41b4298b9253dfa3448efbf8577035f7fdc5/stable/prometheus/templates/alertmanager-clusterrolebinding.yaml#L11
https://github.com/helm/charts/blob/25ab41b4298b9253dfa3448efbf8577035f7fdc5/stable/prometheus/templates/pushgateway-clusterrolebinding.yaml#L11

It also gives an unexpected result to the users because they could think that all objects are non namespaced and deploy it to a namespace other than default with `kubectl apply -n ...`. As I said before, the Clusterrolebindings have a required namespace reference in the `subjects` section and if the referenced Serviceaccount is created in a different namespace than "default",  the creation of the Clusterrolebinding would fail or at least not work as expected.

Because of this I cannot imagine, that someone already uses this chart to template it without specifying a namespace and after that, deploying it manually to another namespace because that wouldn't work.

What I came up with, is some sort of middle ground. When another chart includes Prometheus as subchart and wants to deploy it in another namespace than itself, it can specify the `forceNamespace` parameter. The `forceNamespace` parameter always takes precedence over the .Release.Namespace parameter, specified on the commandline via `helm template -n ...`.
If the parameter `forceNamespace` is not specified, it will use the `.Release.Namespace` parameter.
Yes, if not specified, this parameter is "default". Everything else would be inconsistent because we already use that behavior in the Clusterrolebindings of this chart.

/hold cancel

File: stable/prometheus/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus
-version: 11.2.3
+version: 11.2.4
```
Comment (zanhsieh) [inline stable/prometheus/Chart.yaml]: Let's advance this PR as major version update 11.3.0.

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: stable/prometheus/Chart.yaml (hunk 1 above)
Comment (FrediWeber) [reply stable/prometheus/Chart.yaml]: Okay, done.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**